src/app.js

```diff
@@ -39,7 +38,7 @@ export default function (app) {
 
     if (obj = plugin.hooks) {
       Object.keys(obj).forEach(function (key) {
-        hooks[key].push(obj[key])
+        (hooks[key] ? hooks[key] : hooks[key] = []).push(obj[key])
```

Comment:
@lukejacksonn With this, a plugin can now export unknown hooks and they will be initialized to an empty array here. 

Then, `Router` can check if app.hooks.onRoute exists, and if so, call it.